content/docs/i18n.mdx

```diff
@@ -0,0 +1,359 @@
+---
+id: i18n
+title: i18n
+description: Translate jsonforms
+---
+
+import { I18nExample, ValuesTable } from '../../src/components/docs/i18n';
+
+:::note
+
+Please note that the renderer sets themselves are not yet translatable.
+You can find the current status here:
+https://github.com/eclipsesource/jsonforms/issues/1826
+
+:::
+
+The translate functionality of JSON Forms is integrated into the core component.
+In order to translate JSON Forms, you need to set a translation function and provide it to the JSON Forms component:
+```ts
+const translator = (key, defaultMessage) => {
+  console.log(`Key: ${key}, Default Message: ${defaultMessage}`);
+  return defaultMessage;
+};
+
+const [locale, setLocale] = useState<'de'|'en'>('de');
+const translation = useMemo(() => translator(locale), [locale]);
+
+<JsonForms
+  i18n={{locale: locale, translate: translation}}
+  ...
+/>
+```
+
+The i18n prop consist of three components: `locale`, `translate` and `translateError`. 
+
+## `locale`
+
+Allows to specify the current locale of the application.
+This can be used by renderers to render locale specific UI elements, for example for locale-aware formatting of numbers.
+
+
+## `translate`
+
+Allows to provide a translation function, which handles the actual translation.
+
+:::caution
+
+The translate function should be side effect free and should be stable (memoized) to avoid unnecessary re-renderings, i.e. the translate function should only change if there are new translations.
+
+:::
+
+The type of the translate is
+
+```ts
+(key: string, defaultMessage: string | undefined, context: any) => string | undefined)
+```
+
+and there for has the following parameters:
+
+### `key`
+
+The key is used to identify the string, that needs to be translated.
+It can be set using the `UI Schema`, the `JSON Schema` or is generated by default based on the property path.
+
+#### UI Schema option
+
+The key can be set via the `i18n` UI Schema option:
+```json
+{ 
+  "type": "Control", 
+  "label": "name", 
+  "scope": "#/properties/name", 
+  "options": { 
+    "i18n": "customName" 
+  } 
```

Comment:
This was changed. `"i18n"` is now a direct child property of the Control, i.e. a sibling to `scope`, etc. See https://github.com/eclipsesource/jsonforms/pull/1944